src/mlpack/core/data/detect_file_type.hpp

```diff
@@ -53,7 +53,7 @@ FileType GuessFileType(std::istream& f);
  * @param filename Name of the file.
  * @return The detected file type.  arma::file_type_unknown if unknown.
  */
-FileType AutoDetect(std::fstream& stream,
+inline FileType AutoDetect(std::fstream& stream,
                     const std::string& filename);
```

Comment:
```suggestion
inline FileType AutoDetect(std::fstream& stream,
                           const std::string& filename);
```
I know, it's pedantic. :)